### ISSUE TITLE
Natvis framework cannot display more than 50 content during debugging with ArrayItems or IndexListItems

### ISSUE THREAD

**Issue by atifkarim** [microsoft/vscode-cpptools#9377]
### Environment

- OS and version: Ubuntu, 20.04.4 LTS (Focal Fossa)
- VS Code: 1.67.2
- C/C++ extension: 1.10.3

#### Bug Summary:

The goal is to visualize a stack of memory content (eg: Value from `index 0 to any valid index value` depends on which `size` is given in `<size>size_val</size>` field) using [natvis](https://docs.microsoft.com/en-us/visualstudio/debugger/create-custom-views-of-native-objects?view=vs-2022) which is pointed by a pointer. I have also tried to use `std::vector`. But during debugging the visualizer can show only first `50 entry (index 0 to 49)`. I have used `ArrayItems` and `IndexListItems` but both is giving the same result.

#### Steps to reproduce:

A dummy code I have included here

`foo.h`
```cpp
#include <iostream>
#include <vector>

class Foo
{
    public:
        Foo(){}

        uint32_t *pointer_array;
        std::vector<uint32_t> foo_vec;
};
```
`main.cpp`
```cpp
# include "foo.h"
# define ARRAY_SIZE 5000

int main()
{
    Foo obj_1;

    uint32_t foo_array[ARRAY_SIZE];
    for(int i = 0; i < ARRAY_SIZE; i++)
    {
        foo_array[i] = i*2;
    }
    obj_1.pointer_array = foo_array;

    for(uint32_t i = 0; i < ARRAY_SIZE; i++)
    {
        obj_1.foo_vec.push_back(i*3);
    }

    return 0;
}
```

### Debugger Configurations

The following `natvis file` I have used.

<?xml version="1.0" encoding="utf-8"?>
<AutoVisualizer xmlns="http://schemas.microsoft.com/vstudio/debugger/natvis/2010">
    <Type Name="Foo">
        <DisplayString>Testing_Natvis</DisplayString>
        <Expand>
        <ArrayItems>
            <Size>5000</Size>
            <ValuePointer>pointer_array</ValuePointer>
        </ArrayItems>

        
        

          
          

          
        </Expand>
    </Type>
</AutoVisualizer>

In the `launch.json` I have added extra only the following two lines
```xml
"visualizerFile": "${workspaceFolder}/natvis_file/file.natvis",
"showDisplayString": true,
```

### Debugger Logs
- For better understanding I am giving here a [screenshot](https://imgur.com/NPbCHKf) of the output where in natvis file I have used `IndexListItems` and given size `80` to see value from `index 0 to 79` but the displayed last value is from `index 49`.
- And the [following image](https://imgur.com/NoXt8bj) is showing that I have given the `size` value 6 and `natvis` perfectly is showing value from `index 0 to 5`.

If any workaround is guided to achieve all entry of the memory using Natvis, it would be very beneficial for me.

**Comment by WardenGnaw** [comment]
Known feature request but is tracked in https://github.com/microsoft/MIEngine/issues/821

**Comment by sean-mcmanus** [comment]
Fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/v1.12.2

**Comment by atifkarim** [comment]
@sean-mcmanus  @WardenGnaw 
Thanks for the step to fix the issue. But still I cannot observe more than `1000` entry using Natvis. Each chunk is showing 50 elements and by clicking `More` I can reach max index of `999`. Is it still an intended design limitation, or something else?

**Comment by WardenGnaw** [comment]
> @sean-mcmanus @WardenGnaw Thanks for the step to fix the issue. But still I cannot observe more than `1000` entry using Natvis. Each chunk is showing 50 elements and by clicking `More` I can reach max index of `999`. Is it still an intended design limitation, or something else?

Thanks for letting us know. Based on the engine implementation there is not a limit, but let me double check again.
This may be a limitation on VS Code locals window.

**Comment by WardenGnaw** [comment]
@atifkarim Can you open a new ticket with your issue and potentially provide reproduceable steps?

I was able to go past 1000.
![image](https://user-images.githubusercontent.com/3953714/187007258-5c71e0a8-f13f-4a72-83ae-0b672d672a74.png)

**Comment by atifkarim** [comment]
@WardenGnaw I have just checked that using `IndexListItems` I can observe more than `1000` value. But still `ArrayItems` is limited to `1000` value (0 to 999 index at a time). I have the following questions:
- In your case, have you also worked with `IndexListItems`?
- Do you think `IndexListItems` and `ArrayItems` will provide the same output?

**Comment by WardenGnaw** [comment]
@atifkarim Yep that seems to be an issue. Thank you for reporting it. 

It will be tracked at https://github.com/microsoft/vscode-cpptools/issues/9801

**Comment by sean-mcmanus** [comment]
@atifkarim The 1000 max issue is fixed with 1.13.1: https://github.com/microsoft/vscode-cpptools/releases/tag/v1.13.1
